File: plugin-core/src/main/java/appland/remote/StartRemoteRecordingForm.java
package appland.remote;

import appland.AppMapBundle;
import com.intellij.ui.TextFieldWithHistory;
import com.intellij.ui.components.JBLabel;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;
import java.util.List;

public class StartRemoteRecordingForm {
    private JPanel mainPanel;
    private TextFieldWithHistory urlComboBox;
    private JBLabel infoLabel;
    private JBLabel urlLabel;

    public StartRemoteRecordingForm(@NotNull List<String> recentURLs) {
        urlLabel.setText(AppMapBundle.get("appMapRemoteRecording.urlLabel"));
        infoLabel.setText(AppMapBundle.get("appMapRemoteRecording.message"));
        urlComboBox.setHistory(recentURLs);
    }

    public JPanel getMainPanel() {
        return mainPanel;
    }

    @NotNull
    public String getURL() {
        return this.urlComboBox.getText();
    }

    public TextFieldWithHistory getUrlComboBox() {
        return urlComboBox;
    }
}

File: plugin-core/src/main/kotlin/appland/remote/StartRemoteRecordingForm.kt
package appland.remote

import appland.AppMapBundle
import com.intellij.ui.TextFieldWithHistory
import com.intellij.ui.dsl.builder.AlignX
import com.intellij.ui.dsl.builder.panel
import javax.swing.JPanel

class StartRemoteRecordingForm(recentUrls: List<String>) {
    lateinit var urlComboBox: TextFieldWithHistory
        private set

    val url: String
        get() {
            return urlComboBox.text
        }

    val mainPanel: JPanel = panel {
        row(label = AppMapBundle.get("appMapRemoteRecording.urlLabel")) {
            urlComboBox = textFieldWithHistory(recentUrls)
                .align(AlignX.FILL)
                .focused()
                .component
        }
        row {
            comment(AppMapBundle.get("appMapRemoteRecording.message"))
        }
    }
}

